aea/agent_loop.py

```diff
@@ -166,11 +200,58 @@ def __init__(
 
         :param agent: AEA instance
         :param loop: asyncio loop to use. optional
+        :param threaded: is a new thread to be started for the agent loop
         """
         super().__init__(agent=agent, loop=loop, threaded=threaded)
         self._agent: AbstractAgent = self._agent
 
         self._periodic_tasks: Dict[Callable, PeriodicCaller] = {}
+        self._skill2skill_message_queue: Optional[asyncio.Queue] = None
+
+    def _setup(self) -> None:
+        """Set up agent loop before started."""
+        self._skill2skill_message_queue = asyncio.Queue()
+        super()._setup()
+
+    @property
+    def skill2skill_queue(self) -> Queue:
+        """Get skill to skill message queue."""
+        if not self._skill2skill_message_queue:  # pragma: nocover
```

Comment:
does this work, even if the queue is empty?

Reply:
what I mean is: the condition should evaluate to true only if it is `None`, and not if it is empty

Reply:
it works ok, queue has special method 'empty' for this